sdk/python/tests/online_read_write_test.py

```diff
@@ -91,3 +91,95 @@ def _driver_rw_test(event_ts, created_ts, write, expect_read):
         write=(96864, "I HAVE A NEWER created_ts SO I WIN"),
         expect_read=(96864, "I HAVE A NEWER created_ts SO I WIN"),
     )
+
+
+def basic_feature_service_rw_test(
+    store: FeatureStore, view_name: str, feature_service_name: str
+) -> None:
+    """
+    This is a provider-independent test suite for reading and writing from the online store, to
+    be used by provider-specific tests.
+    """
+    table = store.get_feature_view(name=view_name)
+
+    provider = store._get_provider()
+
+    entity_key = EntityKeyProto(
+        join_keys=["driver"], entity_values=[ValueProto(int64_val=1)]
+    )
+
+    entity_dict = {"driver": 1}
+
+    def _driver_rw_test(event_ts, created_ts, write, expect_read):
+        """ A helper function to write values and read them back """
+        write_lat, write_lon = write
+        expect_lat, expect_lon = expect_read
+        provider.online_write_batch(
+            config=store.config,
+            table=table,
+            data=[
+                (
+                    entity_key,
+                    {
+                        "lat": ValueProto(double_val=write_lat),
+                        "lon": ValueProto(string_val=write_lon),
+                    },
+                    event_ts,
+                    created_ts,
+                )
+            ],
+            progress=None,
+        )
+
+        feature_service = store.get_feature_service(feature_service_name)
+        features = store.get_online_features(
+            features=feature_service, entity_rows=[entity_dict]
+        ).to_dict()
+        assert len(features["driver"]) == 1
+        assert features["driver_locations__lon"][0] == expect_lon
+        assert abs(features["driver_locations__lat"][0] - expect_lat) < 1e-6
+
+    """ 1. Basic test: write value, read it back """
+
+    time_1 = datetime.utcnow()
+    _driver_rw_test(
+        event_ts=time_1, created_ts=time_1, write=(1.1, "3.1"), expect_read=(1.1, "3.1")
+    )
+
+    # Note: This behavior has changed for performance. We should test that older
+    # value can't overwrite over a newer value once we add the respective flag
+    """ Values with an older event_ts should overwrite newer ones """
```

Comment:
It's not clear how these tests relate to feature services.

Reply:
These don't, I split out the changes into a diff file. I'll clean this old one up.

Reply:
Ok.

Reply:
So the changes here have to do with the feature service with link 134-137 where we use the feature service name to get the online features.

The rest of the test is pretty much the same as the existing `test_cli_local`.

Reply:
I guess it's not clear to me why we need to run all the same granular tests for feature services, and if we DO think it's a good idea then wouldn't it make sense to parameterize the other test with a conditional logic (one for feature service and the other for feature refs)?

Reply:
Good idea. I paramterized this test method and removed the dupe